deployment-scanner/src/test/java/org/jboss/as/server/deployment/scanner/FileSystemDeploymentServiceUnitTestCase.java

```diff
@@ -495,6 +495,7 @@ public void testCleanSpuriousMarkers() throws Exception {
         f1 = createFile("spurious" + FileSystemDeploymentService.DEPLOYED);
         f2 = createFile(new File(tmpDir, "nested"), "nested" + FileSystemDeploymentService.DEPLOYED);
 
+        ts.controller.addCompositeSuccessResponse(1);
```

Comment:
I would prefer this test being left as is and one or two next test methods specifically targeting this JIRA be added. It's too hard to reason about how this test change means the fix is being tested.

If you had to do this just to get the existing test to pass, a code comment explaining why it's needed would be good.

Reply:
Hello @bstansberry. Thank you very much for the review and detailed description of the problem.
Yes, I just tried to get the existing test pass.
If I understand it correctly, the test was failing because I added UndeploymentTask in FileSystemDeploymentService which wasn't considered by the mocked controller. I guess the addCompositeSuccessResponse method only adds one success response to MockServerController.responses to use it in the processOp method somehow as a counter of requests.
Should I also try to create some other test?
For now I will force-push my changes.

Reply:
Hi @khermano -- yes, please add another test or two, e.g. one that specifically covers the bug and another that validates that a deployment not associated with the scanner being tested isn't undeployed just because a cruft FAILED_DEPLOY marker file is left in the scanner's directory. I *think* the test infrastructure used by FileSystemDeploymentServiceTestCase can handle that scenario, or at least is close to being able to.

It sounds like the line I commented on here needs to remain, which is fine, but preceding it with a brief comment would be good as it's not obvious why it's there.

Reply:
Hello @bstansberry , 
so I added the comment and the test for the bug you recommended, but I am unsure about the other one.
I tried a couple of things to deploy the failing kitchensink.war through CLI and console, but none of them create any content in the scanned standalone/deployments directory, not even `deploy ~/../kitchensink.war --force`.
And I think that maybe creating markers like ".failed", ".deployed" and others is only a deployment-scanner thing, or am I wrong about that? 

Reply:
Hi @bstansberry ,
I am sorry for the delay. I got stuck with this and needed to work on something else, but I think I finally have it. 
So again, I'm really sorry for the inconvenience. Hopefully, it looks okay.